exist-core/src/test/java/org/exist/xquery/FunctionTypeInElementContent.java

```diff
@@ -0,0 +1,174 @@
+     */
+    @Ignore
+    @Test
+    public void functionVariable() throws XMLDBException {
+        final String query = "let $f := function () { () } return element test { $f }";
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got function(*) [at line 1, column 16, source: element test { sum(?) }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * TODO: remove empty sequence after https://github.com/eXist-db/exist/issues/3551 is fixed
+     */
+    @Test
+    public void userDefinedFunction() throws XMLDBException {
+        final String query = "element test { function () { () } }";
+        // TODO: user defined function has its location offset to a weird location
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got function(*) [at line 1, column 25, source: element test { function () { () } }]";
+        assertCompilationError(query, error);
+    }
+
+    @Test
+    public void mapConstructor() throws XMLDBException {
+        final String query = "element test { map {} }";
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got map(*) [at line 1, column 16, source: element test { map {} }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * sequence in enclosed expression with only a function type
+     */
+    @Ignore
+    @Test
+    public void sequenceOfMaps() throws XMLDBException {
+        final String query = "element test { (map {}) }";
+        final String error = "An exception occurred during query execution: err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element foo { (map{}) }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * there is an edge case which would evaluate to empty sequence
+     * but should arguably still throw
+     * Does still throw without location info
+     * TODO: add (sub-expression) location
+     */
+    @Test
+    public void sequenceOfMapsEdgeCase() throws XMLDBException {
+        final String query = "element test { (map {})[2] }";
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element test { (map {})[2] }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * -- no error is thrown at compile time
+     * TODO: add (sub-expression) location
+     */
+    @Test
+    public void ArrayOfMaps() throws XMLDBException {
+        final String query = "element test { [map {}] }";
+        final String error = "An exception occurred during query execution: err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element test { [map{}] }]";
+        assertCompilationError(query, error);
+    };
+
+    /**
+     * This could throw at compile time, but does not
+     * TODO: add (sub-expression) location
+     */
+    @Ignore
```

Comment:
What is the status/impact of this ignored test and TODO? do you mean to fix it in this PR?

Reply:
see above